src/containerapp/azext_containerapp/custom.py

```diff
@@ -881,6 +886,10 @@ def update_containerapp_logic(cmd,
         new_containerapp["properties"]["template"] = {} if "template" not in new_containerapp["properties"] else new_containerapp["properties"]["template"]
         new_containerapp["properties"]["template"]["revisionSuffix"] = revision_suffix
 
+    if termination_grace_period is not None:
+        new_containerapp["properties"]["template"] = {} if "template" not in new_containerapp["properties"] else new_containerapp["properties"]["template"]
```

Comment:
Recommend to use `safe_set("properties", "template", "terminationGracePeriodSeconds", value=termination_grace_period)`